dessia_common/excel_reader.py

```diff
@@ -0,0 +1,482 @@
+        """
+        init_signature = inspect.signature(obj_class.__init__)
+        init_parameters = init_signature.parameters
+
+        attribute_info = {}
+
+        for attr_name, param in init_parameters.items():
+            if attr_name != 'self':
+                attr_info = {
+                    'type': param.annotation if param.annotation != param.empty else None,
+                    'default_value': param.default if param.default != param.empty else "empty"
+                }
+                attribute_info[attr_name] = attr_info
+
+        return attribute_info
+
+    @staticmethod
+    def get_data(value, attributes, init_attributes):
+        """
+        Extracts relevant data based on attributes and initial attributes for object instantiation.
+
+        Args:
+        - value (list): List containing values corresponding to attributes.
+        - attributes (list): List of attributes related to the object.
+        - init_attributes (list): List of initial attributes for object instantiation.
+
+        Returns:
+        - dict: Dictionary containing extracted object data based on matching attributes and initial attributes.
+        """
+        check_missed_attribute, attribute = missed_attribute(attributes, init_attributes)
+        if not check_missed_attribute:
+            raise ValueError(f"missing attribute '{attribute}' don't have a default value")
+
+        object_data = {}
+        for i, attribute in enumerate(attributes):
+            if attribute in init_attributes:
+                if isinstance(value[i], str):
+                    try:
+                        object_data[attribute] = literal_eval(value[i])
+                    except (ValueError, SyntaxError):
+                        object_data[attribute] = value[i]
+                else:
+                    object_data[attribute] = value[i]
+        return object_data
+
+    def instantiate_main_objects(self, instantiated_objects, key, values):
+        """
+        Instantiate main objects based on provided values and attributes.
+
+        Args:
+        - instantiated_objects_list (dict): Dictionary containing instantiated objects.
+        - key (str): Key to identify the instantiated objects.
+        - values (list): List containing module name, class name, and attributes.
+
+        Returns:
+        - dict: Updated dictionary of instantiated objects.
+        """
+
+        class_info = {'module_name': values[0][0], 'class_name': values[0][1], 'object_attributes': values[1]}
+        full_class_name = f"{class_info['module_name']}.{class_info['class_name']}"
+        obj_class = get_python_class_from_class_name(full_class_name=full_class_name)
+
+        initial_attributes = self.get_attributes_and_types(obj_class)
+        objects = []
+
+        for value_set in values[2].values():
+            for index, attr_value in enumerate(value_set):
+                if isinstance(attr_value, openpyxl.cell.cell.Cell):
```

Comment:
What happens if it is not a Cell ?

Reply:
if cell, We compute the real value, if not we keep the value (exemple: int, str ....)